tests/test_coverage_test.ts

```diff
@@ -0,0 +1,48 @@
+import { execSync } from "child_process";
+import fs from "fs";
+import path from "path";
+import { REPO_ROOT } from "tests/utils";
+
+// TODO(Tyler): Burndown this list (currently all linters defined before the testing framework was implemented).
+const excludedLinters: string[] = ["codespell", "cspell", "nancy", "oxipng", "sqlfmt", "trivy"];
+
+// This test asserts that all linters have at least one test. All new linters are expected to have
+// test coverage. Review tests/readme.md for testing guidelines. Prefer using npm test for indirection
+// in this test so that we get an accurate list of all tests, regardless of any changes to the test spec
+// in jest.config.ts.
+describe("All linters must have tests", () => {
```

Comment:
nice!

Spent a little time stewing on the dir structure for `tests/` and I think I'm fine with the current state.

* Generally I prefer for tests to be separate from testing utils, but this also results in `tests/` and `testing/` (which has both a tab completion problem and also a "what's the difference" question for newcomers)
* Most testlib/testutil logic is in index.ts or subdirs, so it's not too much of a difference.
* I wasn't super happy with other names that I was coming up with:
  * `self_test/coverage_test.ts`
  * `trunk/coverage_test.ts`
  * `plugins_test/coverage_test.ts`
  * `meta/coverage_test.ts`
  * (obviously you can mix and match `foo/` and `foo_test/` above)
* The only non-test file we have that lives at the root of `tests/` is `index.ts`, which is... fine, I think.

Reply:
Thanks for the thinking here. I suspect we may revisit this a bit in the intermediate future. As for the last bullet point, I liked having the `index.ts` in `tests/`, since it makes imports in the individual linter tests easy--you can just import `linterCheckTest` from `tests` rather than specifying a longer and repetitive path.